skimage/feature/util.py

```diff
@@ -145,22 +147,30 @@ def plot_matched_features(
     ax.imshow(image, cmap='gray')
     ax.axis((0, image0.shape[1] + offset[1], image0.shape[0] + offset[0], 0))
 
-    rng = np.random.default_rng()
+    number_of_matches = matches.shape[0]
 
-    for i in range(matches.shape[0]):
-        idx0 = matches[i, 0]
-        idx1 = matches[i, 1]
+    from matplotlib.colors import is_color_like
 
-        if matches_color is None:
-            color = rng.random(3)
-        else:
-            color = matches_color
+    if matches_color is None:
+        rng = np.random.default_rng(seed=0)
+        colors = [rng.random(3) for _ in range(number_of_matches)]
```

Comment:
`rng.random((number_of_matches, 3))` ?

Also, is it smart to select colors fully randomly? Perhaps we'd be better off selecting randomly from a know palette. But, fine, that's the current (slightly odd) behavior.

Reply:
I guess the number of features factors into this heavily. My instinct here is to chose a good default and let users worry about finding the perfect `matches_color` for their use themselves. 

For that reason I'm also not too fond of the current solution that allows cycling of colors. I don't see a clear advantage that makes it worth expanding our API. Besides making it somewhat easier for users to pass something valid to `matches_color`, it might often lead to worse outcomes than seeded random sampling. It might even hide errors when users accidentally & unknowingly pass a wrong length and don't use this feature intentionally. Users that want that feature can also just use `cycle` themselves to prepare `matches_color` of the correct length.


@michaelbratsch, sorry for chiming in after you made these changes. What do you think?

Reply:
cc @mkcor

Reply:
As just discussed in the community call, I reverted the usage of `cycle` due to the reasons explained in https://github.com/scikit-image/scikit-image/pull/7541#discussion_r1773106310 so that the user needs to pass a single color or a sequence with matching length to `matches_color`.

Reply:
I'm trying to understand the reasoning here. The user, in most cases, will not know how many matches they're going to be working with. For them to match the number of colors to the number of features, without an example of how to do so, will be pretty tricky, and the docs don't currently explain the limitation even? Cycling colors when displaying feature matches feels pretty innocuous; it's not like repetition matters all that much, as long as you have enough colors being cycled through?

I wasn't at the community call this week, so apologies if I missed some nuance.

Reply:
But the length is easily computed once you have the matches right? 

I was trying to think about reasons, for why a user might want to control the colors over random ones. One use case I came up with is, that users might want to color code machtes by some property. So this feature is less about convenience and doing something for the user and more about making it even possible. 

To me, cycling seemed not a good fit for multiple reasons:
- It's "easy" to implement from a user side and the above API with `itertools.cylce`.
- I actually think cycling makes for a more complicated API. We would have to deal gracefully with an empty iterable (exception?) or an iterable with 1 item (warning?).
- I couldn't really think of good use cases for cycling or where it leads to better results than random coloring. I agree with @michaelbratsch sentiment, that it's harder to differentiate the lines with cycling through fewer colors.

So all in all, asking for a sequence of established length, seemed like the more API conservative and less assuming choice. 

Does that change your perspective? Am I missing something?

Reply:
We could just cover `cycle(matches_color)` in an example. But what is the problem with cycling over a list that is already the right length? It doesn't cause any difficulty. It does, however, address the common case where you have a palette of N colors, and an as-yet-to-be-determined number of matches.

Reply:
> But what is the problem with cycling over a list that is already the right length? [...] It does, however, address the common case where you have a palette of N colors, and an as-yet-to-be-determined number of matches.

There's no problem and true. But I see other drawbacks listed above. Do you disagree with them or do you assign a different weight to them? Otherwise, I'd personally, just cover the cycling case in a docstring example. 

I also don't want to ask @michaelbratsch to revert the reversion. :sweat_smile: